PushApp.ts

```diff
@@ -0,0 +1,125 @@
+import {
+    IAppAccessors,
+    ILogger,
+    IConfigurationExtend,
+    IEnvironmentRead,
+} from '@rocket.chat/apps-engine/definition/accessors';
+import { App } from '@rocket.chat/apps-engine/definition/App';
+import { IAppInfo } from '@rocket.chat/apps-engine/definition/metadata';
+import { SettingType } from '@rocket.chat/apps-engine/definition/settings';
+import { Test } from './commands/Test';
+import { ApiVisibility, ApiSecurity } from '@rocket.chat/apps-engine/definition/api';
+import { RocketEndpoint } from './endpoints/RocketEndpoint';
+import { PushEndpoint } from './endpoints/PushEndpoint';
+
+export class PushApp extends App {
+    constructor(info: IAppInfo, logger: ILogger, accessors: IAppAccessors) {
+        super(info, logger, accessors);
+
+    }
+
+    async extendConfiguration(configuration: IConfigurationExtend, environmentRead: IEnvironmentRead): Promise<void> {
+
+        await configuration.api.provideApi({
+            visibility: ApiVisibility.PUBLIC,
+            security: ApiSecurity.UNSECURE,
+            endpoints: [
+                new RocketEndpoint(this),
+                new PushEndpoint(this)
+            ]
+        })
+
+        await configuration.settings.provideSetting({
+            id:  "rc_ptkn",
+            type: SettingType.STRING,
+            packageValue: '',
+            required: true,
+            public: false,
+            i18nLabel: 'Rocketchat Personal Access Token',
```

Comment:
`i18nLabel` is the reference to JSON files at /i18n folder for the app translations, see an example:
https://github.com/push-platform/Rocket.Chat.App-QuickMessage/tree/master/i18n